web/src/js/components/General/EmailField.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import TextField from '@material-ui/core/TextField'
+
+class EmailField extends React.Component {
+  constructor(props) {
+    super(props)
+    this.inputElem = null
+    this.state = {
+      email: null,
+      validationErrorMessage: null,
+    }
+  }
+
+  hasValue() {
+    return !!(
+      this.inputElem &&
+      this.inputElem.value &&
+      this.inputElem.value.trim()
+    )
+  }
+
+  getValue() {
+    if (this.hasValue()) {
+      return this.inputElem.value.trim()
+    }
+    return null
+  }
+
+  setErrorMessage(message) {
+    this.setState({
+      validationErrorMessage: message,
+    })
+  }
+
+  validate() {
+    if (this.hasValue()) {
```

Comment:
we may want to add a basic email validation library or regex on the front end
something that checks that the email has an @ and a .x